mantidproject/mantid
# Indirect - Fix Missing Logs in Edit Local Fit Parameters Dialog

**Description of work.**
This PR fixes an issue in Indirect Data Analysis where the logs were not being loaded in the `Edit Local Fit Parameters` dialog.

This is because the `FunctionModel` assumes a dataset (i.e. workspace) contains only one spectrum to be fitted. This is true for Muon Analysis it seems, but not for Indirect.

To fix this, the `FunctionModel` now stores a QMap containing workspace names and the spectra corresponding to these workspaces. Making this change allows the retrieval of dataset names and domain names from the `FunctionModel`. By using both of these, sample logs for multi-spectra workspaces can now be loaded by the `Edit Local Fit Parameters` dialog.

**To test:**
1. `Interfaces`->`Indirect`->`Data Analysis`
2. Go to `IqtFit` tab
3. Load the data below
4. Select Simultaneous Fit
5. Select 1 exponential
6. Tick `See Full Function`
7. Click the `...` box next to one of the parameters
8. The Log names should be loaded in the combobox at the bottom of the dialog.

**Data**
[irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/5428108/irs26176_graphite002_iqt.zip)

Make sure you can add and remove datasets OK on the Muon Analysis interface.

Fixes #29802

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
